### ISSUE TITLE
running RLlib assumes cuda and crashes

### ISSUE THREAD

**Issue by worldveil** [ray-project/ray#20171]
I've noticed this before, but we should give users a better error

```
% git clone git@github.com:anyscale/cuj-rl-in-production.git ./repro
% cd repro
% source env.sh error_repro
% ray stop
% ray start --head
% python cuj.py  --as-test

2021-11-08 20:37:09,908 ERROR trial_runner.py:924 -- Trial APPO_MysteriousCorridor_afd8a_00004: Error processing event.
Traceback (most recent call last):
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/tune/trial_runner.py", line 890, in _process_trial
    results = self.trial_executor.fetch_result(trial)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/tune/ray_trial_executor.py", line 788, in fetch_result
    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/worker.py", line 1627, in get
    raise value
ray.exceptions.RayActorError: The actor died because of an error raised in its creation task, ray::APPO.__init__() (pid=21386, ip=127.0.0.1)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/agents/trainer_template.py", line 137, in __init__
    Trainer.__init__(self, config, env, logger_creator)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 623, in __init__
    super().__init__(config, logger_creator)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/tune/trainable.py", line 107, in __init__
    self.setup(copy.deepcopy(self.config))
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/agents/trainer_template.py", line 147, in setup
    super().setup(config)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 776, in setup
    self._init(self.config, self.env_creator)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/agents/trainer_template.py", line 171, in _init
    self.workers = self._make_workers(
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 858, in _make_workers
    return WorkerSet(
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/evaluation/worker_set.py", line 87, in __init__
    remote_spaces = ray.get(self.remote_workers(
ray.exceptions.RayActorError: The actor died because of an error raised in its creation task, ray::RolloutWorker.__init__() (pid=21418, ip=127.0.0.1)
  File "/Users/will/opt/anaconda3/envs/cuj_rl_new2/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 567, in __init__
    devices = list(range(torch.cuda.device_count()))
AttributeError: 'NoneType' object has no attribute 'cuda'
```

this is fixed with:

```
pip install tensorflow torch
python cuj.py --as-test
```

**Comment by sven1977** [comment]
Yeah, @avnishn is working on these kinds of failure-modes and better error messages (e.g. no DL framework installed). Assigning him here.

**Comment by avnishn** [comment]
I believe this is closed by #20205
